tests/host/common/Arduino.cpp

```diff
@@ -53,10 +54,33 @@ extern "C" void optimistic_yield (uint32_t interval_us)
     (void)interval_us;
 }
 
+extern "C" void esp_suspend()
+{
+}
+
+extern "C" void esp_schedule()
+{
+}
+
 extern "C" void esp_yield()
 {
 }
 
+extern "C" void esp_delay (unsigned long ms)
+{
+    usleep(ms * 1000);
+}
+
+bool esp_try_delay(const uint32_t start_ms, const uint32_t timeout_ms, const uint32_t intvl_ms) {
+    decltype(millis()) expired;
+
+    if ((expired = millis() - start_ms) >= timeout_ms) {
+        return true;
+    }
+    const auto remaining = timeout_ms - expired;
+    esp_delay(remaining <= intvl_ms ? remaining : intvl_ms);
+    return false;
+}
```

Comment:
revisiting the tests, nothing actually calls it :/
`return true;` might suffice for now (...and if it's not working, host test Checks will show)

Reply:
I agree with your remarks regarding esp_yield() in startWaveform; done.
I don't see how ignoring the specification in the comment is the right conclusion from any absence of a test :-)
Careful examination of the code in review hopefully is sufficient for this change.

Reply:
> I don't see how ignoring the specification in the comment is the right conclusion from any absence of a test :-)

Now there are two implementations though, one in core\_esp8266\_main.cpp and another esp\_try\_delay is here. The way tests work right now, there's no async waiting like here since the rest of the mock code makes blocking operations instead. So that is why my first though is to just remove the implementation until it is actually doing something useful.

Another option is to actually share the code, however that might happen (actually build Core's .cpp, extra include with implementation, macro resulting in the implementation code, etc.)

> // No mem barrier here, the call to a global function implies global state updated

Was the comment wrong?

Reply:
The host-test version of `esp_try_delay` is identical to the main one, now. Does no harm this way, despite you are right that at present, there cannot be any async cancellation.
Fixed the comment specifying `esp_try_delay`.
The memory fence in the PWM routine is safer for any code changes, like inlining, at some point in time, I don't think such micro-optimizations by hand are a great idea for stability.